arrow-libs/optics/arrow-optics/build.gradle.kts

```diff
@@ -20,6 +20,10 @@ if (enableCompatibilityMetadataVariant) {
   }
 }
 
+tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
+  kotlinOptions.freeCompilerArgs += listOf("-Xenable-builder-inference", "-opt-in=kotlin.RequiresOptIn")
```

Comment:
Are both these flags needed?

First one is to allow builder inference inside Arrow, or to be able to expose annotation?
Second one is to allow OptIn without a compiler warning?

I’m always confused about the OptIn, because it works fine without this flag.

Reply:
I am not sure, the docs at https://kotlinlang.org/docs/using-builders-with-builder-inference.html#writing-your-own-builders are by no means clear. But I'll delete the opt-in one.

Reply:
> But I'll delete the opt-in one.

I'm not sure if we need it yes, or no 😅 Just wanted to clarify, it can be that we're currently missing it.

Reply:
As you said "are by no means clear" :/ I feel the same for the `opt-in` one.